File: 1768-merge-strings-alternately/main.go
func mergeAlternately(word1 string, word2 string) string {
    
    l1, l2 := len(word1), len(word2)
    minLength := min(l1, l2)

    var ans string = ""
    for i := range minLength {
        ans += string(word1[i]) + string(word2[i])
    }

    if l1 > l2 {
        ans += word1[minLength:]
    }else {
        ans += word2[minLength:]
    }

    return ans
}

func min(x, y int) int {
    if x > y {
        return y
    }
    return x
}

File: 1768-merge-strings-alternately/main.rs
impl Solution {
    pub fn merge_alternately(word1: String, word2: String) -> String {
        
        let min_length = word1.len().min(word2.len());
        
        let mut ans = String::new();
        for (c1, c2) in word1.chars().zip(word2.chars()) {
            ans.push(c1);
            ans.push(c2);
        }

        ans.push_str(&word1[min_length..]);
        ans.push_str(&word2[min_length..]);

        ans
    }
}
